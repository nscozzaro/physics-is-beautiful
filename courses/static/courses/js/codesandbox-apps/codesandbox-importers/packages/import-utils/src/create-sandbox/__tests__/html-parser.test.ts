import parser from "../html-parser";

describe("html-parser", () => {
  it("can retrieve body from html", () => {
    const BODY_HTML = `
      <!doctype html>
      <html lang="en">
        <head>
          <meta charset="utf-8">
          <meta name="viewport" content="width=device-width, initial-scale=1">
          <title>Redux Shopping Cart Example</title>
        </head>
        <body>
          <div id="root"></div>
          <!--
            This HTML file is a template.
            If you open it directly in the browser, you will see an empty page.
            You can add webfonts, meta tags, or analytics to this file.
            The build step will place the bundled scripts into the <body> tag.
            To begin the development, run \`npm start\` in this folder.
            To create a production bundle, use \`npm run build\`.
          -->
        </body>
      </html>
    `;
    expect(parser(BODY_HTML)).toMatchSnapshot();
  });

  it("can retrieve js external resources", () => {
    const BODY_HTML = `
      <!doctype html>
      <html lang="en">
        <head>
          <meta charset="utf-8">
          <meta name="viewport" content="width=device-width, initial-scale=1">
          <title>Redux Shopping Cart Example</title>
          <link href="https://redux-form.com/6.8.0/bundle.css"
            media="screen, projection"
            rel="stylesheet" type="text/css"/>
          <link href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.3.0/css/font-awesome.min.css"
            media="screen, projection" rel="stylesheet" type="text/css"/>
        </head>
        <body>
          <div id="root"></div>
          <!--
            This HTML file is a template.
            If you open it directly in the browser, you will see an empty page.
            You can add webfonts, meta tags, or analytics to this file.
            The build step will place the bundled scripts into the <body> tag.
            To begin the development, run \`npm start\` in this folder.
            To create a production bundle, use \`npm run build\`.
          -->
          <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.slim.min.js"></script>
        </body>
      </html>
    `;

    expect(parser(BODY_HTML)).toMatchSnapshot();
  });

  it("can retrieve css external resources", () => {
    const BODY_HTML = `
      <!doctype html>
      <html lang="en">
        <head>
          <meta charset="utf-8">
          <meta name="viewport" content="width=device-width, initial-scale=1">
          <title>Redux Shopping Cart Example</title>
          <link href="https://redux-form.com/6.8.0/bundle.css"
            media="screen, projection"
            rel="stylesheet" type="text/css"/>
          <link href="//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.3.0/css/font-awesome.min.css"
            media="screen, projection" rel="stylesheet" type="text/css"/>
        </head>
        <body>
          <div id="root"></div>
          <!--
            This HTML file is a template.
            If you open it directly in the browser, you will see an empty page.
            You can add webfonts, meta tags, or analytics to this file.
            The build step will place the bundled scripts into the <body> tag.
            To begin the development, run \`npm start\` in this folder.
            To create a production bundle, use \`npm run build\`.
          -->
        </body>
      </html>
    `;

    expect(parser(BODY_HTML)).toMatchSnapshot();
  });
});
