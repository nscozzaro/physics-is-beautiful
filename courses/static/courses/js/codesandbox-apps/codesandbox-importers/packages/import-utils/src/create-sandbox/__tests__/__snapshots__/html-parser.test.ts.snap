// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html-parser can retrieve body from html 1`] = `
Object {
  "body": "
          <div id=\\"root\\"></div>
          <!--
            This HTML file is a template.
            If you open it directly in the browser, you will see an empty page.
            You can add webfonts, meta tags, or analytics to this file.
            The build step will place the bundled scripts into the <body> tag.
            To begin the development, run \`npm start\` in this folder.
            To create a production bundle, use \`npm run build\`.
          -->
        ",
  "externalResources": Array [],
}
`;

exports[`html-parser can retrieve css external resources 1`] = `
Object {
  "body": "
          <div id=\\"root\\"></div>
          <!--
            This HTML file is a template.
            If you open it directly in the browser, you will see an empty page.
            You can add webfonts, meta tags, or analytics to this file.
            The build step will place the bundled scripts into the <body> tag.
            To begin the development, run \`npm start\` in this folder.
            To create a production bundle, use \`npm run build\`.
          -->
        ",
  "externalResources": Array [
    "https://redux-form.com/6.8.0/bundle.css",
    "//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.3.0/css/font-awesome.min.css",
  ],
}
`;

exports[`html-parser can retrieve js external resources 1`] = `
Object {
  "body": "
          <div id=\\"root\\"></div>
          <!--
            This HTML file is a template.
            If you open it directly in the browser, you will see an empty page.
            You can add webfonts, meta tags, or analytics to this file.
            The build step will place the bundled scripts into the <body> tag.
            To begin the development, run \`npm start\` in this folder.
            To create a production bundle, use \`npm run build\`.
          -->
          <script type=\\"text/javascript\\" src=\\"https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.slim.min.js\\"></script>
        ",
  "externalResources": Array [
    "https://redux-form.com/6.8.0/bundle.css",
    "//cdnjs.cloudflare.com/ajax/libs/font-awesome/4.3.0/css/font-awesome.min.css",
    "https://cdnjs.cloudflare.com/ajax/libs/jquery/3.2.1/jquery.slim.min.js",
  ],
}
`;
